pymeasure/instruments/keithley/keithley6487.py

```diff
@@ -0,0 +1,395 @@
+    # Filters #
+    ###########
+
+    filter_type = Instrument.control(
+        ":SENS:AVER:TCON?", ":SENS:AVER:TCON %s",
+        """ A String property that controls the filter's type.
+        REP : Repeating filter
+        MOV : Moving filter""",
+        validator=strict_discrete_set,
+        values=['REP', 'MOV'],
+        map_values=False)
+
+    filter_count = Instrument.control(
+        ":SENS:AVER:COUNT?", ":SENS:AVER:COUNT %d",
+        """ A integer property that controls the number of readings that are
+        acquired and stored in the filter buffer for the averaging""",
+        validator=truncated_range,
+        values=[2, 100],
+        cast=int)
+
+    filter_state = Instrument.control(
+        ":SENS:AVER?", ":SENS:AVER %s",
+        """ A string property that controls if the filter is active.""",
+        validator=strict_discrete_set,
+        values=['ON', 'OFF'],
+        map_values=False)
+
+    ####################
+    #   Methods        #
+    ####################
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter, "Keithley 6487 Picoammeter", **kwargs
+        )
+
+    def enable_source(self):
+        """Enables the source voltage of the instrument"""
+        self.write(":SOUR:VOLT:STAT ON")
+
+    def disable_source(self):
+        """Disables the source voltage of the instrument"""
+        self.write(":SOUR:VOLT:STAT OFF")
+
+    def measure_current(self, nplc=1, current=1.05e-4, auto_range=True):
+        """ Configures the measurement of current.
+
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 10
+        :param current: Upper limit of current in Amps, from -0.021 A to 0.021 A
+        :param auto_range: Enables auto_range if True, else uses the set current
+        """
+        log.info("%s is measuring current." % self.name)
+        self.write(":SENS:FUNC:CURR;"
+                   ":SENS:CURR:NPLC %f;:FORM:ELEM READ;" % nplc)
+        if auto_range:
+            self.write(":SENS:CURR:RANG:AUTO 1;")
+        else:
+            self.current_range = current
+        self.check_errors()
+
+    def auto_range_current(self):
+        """ Configures the current measurement to use an automatic range.
+        """
+        self.write(":SENS:CURR:RANG:AUTO 1")
+
+    display_enabled = Instrument.control(
+        ":DISP:ENAB?", ":DISP:ENAB %d",
+        """ A boolean property that controls whether or not the display of the
+        sourcemeter is enabled. Valid values are True and False. """,
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+
+    @property
+    def error(self):
+        """ Returns a tuple of an error code and message from a
+        single error. """
+        err = self.values(":system:error?")
+        if len(err) < 2:
+            err = self.read()  # Try reading again
+        code = err[0]
+        message = err[1].replace('""', '')
+        return (code, message)
+
+    def check_errors(self):
```

Comment:
`check_errors` is already available at `Instrument` class level.
Is there any reason to redefine it ?